File: java/aws/lambda/src/main/java/example/MockContext.java
package example;

import com.amazonaws.services.lambda.runtime.ClientContext;
import com.amazonaws.services.lambda.runtime.CognitoIdentity;
import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.LambdaLogger;

public class MockContext implements Context {

    @Override
    public String getAwsRequestId() {
        return "mockRequestId";
    }

    @Override
    public String getLogGroupName() {
        return "mockLogGroupName";
    }

    @Override
    public String getLogStreamName() {
        return "mockLogStreamName";
    }

    @Override
    public String getFunctionName() {
        return "mockFunctionName";
    }

    @Override
    public String getFunctionVersion() {
        return "mockFunctionVersion";
    }

    @Override
    public String getInvokedFunctionArn() {
        return "mockFunctionArn";
    }

    @Override
    public CognitoIdentity getIdentity() {
        return null;
    }

    @Override
    public ClientContext getClientContext() {
        return null;
    }

    @Override
    public int getRemainingTimeInMillis() {
        return 300000; // 5 minutes
    }

    @Override
    public int getMemoryLimitInMB() {
        return 512; // Mock 512MB memory
    }

    @Override
    public LambdaLogger getLogger() {
        return new MockLambdaLogger();
    }
}

File: java/aws/lambda-kotlin/src/test/kotlin/example/invoke/MockContext.kt
package example.invoke

import com.amazonaws.services.lambda.runtime.ClientContext
import com.amazonaws.services.lambda.runtime.CognitoIdentity
import com.amazonaws.services.lambda.runtime.Context
import com.amazonaws.services.lambda.runtime.LambdaLogger

class MockContext : Context {

    override fun getAwsRequestId(): String = "mockRequestId"
    override fun getLogGroupName(): String = "mockLogGroupName"
    override fun getLogStreamName(): String = "mockLogStreamName"
    override fun getFunctionName(): String = "mockFunctionName"
    override fun getFunctionVersion(): String = "mockFunctionVersion"
    override fun getInvokedFunctionArn(): String = "mockFunctionArn"
    override fun getIdentity(): CognitoIdentity? = null
    override fun getClientContext(): ClientContext? = null
    override fun getRemainingTimeInMillis(): Int = 300_000
    override fun getMemoryLimitInMB(): Int = 512
    override fun getLogger(): LambdaLogger = MockLambdaLogger()
}
